[MIG] connector_importer_source_sftp: Migration to 16.0

p-tombez opened the pull request:
ref: BSALLO-1566

simahawk committed:
Add connector_importer_source_sftp

simahawk committed:
connector_importer_source_sftp: migrate to v14

oca-travis committed:
[UPD] Update connector_importer_source_sftp.pot

OCA-git-bot committed:
[UPD] README.rst

damdam-s committed:
c_importer_source_sftp: fix move file

A file can be moved only if we have a destination path.

OCA-git-bot committed:
connector_importer_source_sftp 14.0.1.1.0

dhx committed:
[MIG] connector_importer_source_sftp: Migration to 15.0

dhx committed:
[IMP] connector_importer_source_sftp: Handle no or empty file found case

Instead of throwing an exception with an unclear message in the _get_line function

Commit:
[UPD] Update connector_importer_source_sftp.pot

OCA-git-bot committed:
[UPD] README.rst

simahawk committed:
connector_importer_source_sftp: fix pin cryptography<37

OCA-git-bot committed:
connector_importer_source_sftp 15.0.1.0.1

Commit:
[UPD] Update connector_importer_source_sftp.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[IMP] connector_importer_source_sftp: pre-commit stuff

Commit:
[MIG] connector_importer_source_sftp: Migration to 16.0

simahawk commented on test-requirements.txt:
```diff
@@ -1 +1,2 @@
 odoo-test-helper
+mock
```
this is not needed. Just change the import of mock to `from unittest import mock` :wink:

p-tombez replied:
It **IS** needed because this depends on `storage_backend_sftp`, which also requires `mock`.
See https://github.com/OCA/connector-interfaces/actions/runs/10612682197/job/29414865086#step:8:209

simahawk replied:
it is **NOT** because that's broken there too. It has been migrated in the wrong way. There's no need for such dep. Trust me :wink:

simahawk replied:
in red the wrong ones 

![image](https://github.com/user-attachments/assets/dd68af7b-2a0b-4c11-b9fa-412c467867ce)

simahawk replied:
There you go https://github.com/OCA/storage/pull/388

simahawk committed:
fixup! [MIG] connector_importer_source_sftp: Migration to 16.0

simahawk commented:
FTR I couldn't replicate the failing test on the event listener locally
```
2024-09-04 05:27:22,457 260 ERROR odoo odoo.tests.suite: ERROR: setUpClass (odoo.addons.connector_importer_source_sftp.tests.test_event_listener.TestRecordImporterFinishedEvent)
Traceback (most recent call last):
  File "/__w/connector-interfaces/connector-interfaces/connector_importer_source_sftp/tests/test_event_listener.py", line 25, in setUpClass
    cls.recordset.write(
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/component_event/models/base.py", line 109, in write
    result = super(Base, self).write(vals)
  File "/opt/odoo/odoo/models.py", line 3784, in write
    field.write(self, value)
  File "/opt/odoo/odoo/fields.py", line 1125, in write
    cache_value = self.convert_to_cache(value, records)
  File "/opt/odoo/odoo/fields.py", line 2735, in convert_to_cache
    raise ValueError("Wrong value for %s: %r" % (self, value))
ValueError: Wrong value for import.recordset.source_model: 'import.source.csv.sftp'
```
Looks weird because the selection option should be already there.

simahawk commented:
Ok. it fails only if you run tests at db creation.

simahawk committed:
c_importer_source_sftp: fix event listeners tests

They must run at post_install to make sure the new source type is registered.

Fixes:

```
odoo.tests.suite: ERROR: setUpClass (odoo.addons.connector_importer_source_sftp.tests.test_event_listener.TestRecordImporterFinishedEvent)
Traceback (most recent call last):
  File "/__w/connector-interfaces/connector-interfaces/connector_importer_source_sftp/tests/test_event_listener.py", line 25, in setUpClass
    cls.recordset.write(
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/component_event/models/base.py", line 109, in write
    result = super(Base, self).write(vals)
  File "/opt/odoo/odoo/models.py", line 3784, in write
    field.write(self, value)
  File "/opt/odoo/odoo/fields.py", line 1125, in write
    cache_value = self.convert_to_cache(value, records)
  File "/opt/odoo/odoo/fields.py", line 2735, in convert_to_cache
    raise ValueError("Wrong value for %s: %r" % (self, value))
ValueError: Wrong value for import.recordset.source_model: 'import.source.csv.sftp'
```

simahawk commented:
Great, this change https://github.com/OCA/connector-interfaces/pull/138/commits/876c6097e3790956e5bdc88b11757d4fb5f6c6b1 fixed tests locally but not here :sweat:

leemannd committed:
IMP connector_importer_product update cryptography dep

leemannd commented:
I added a PR to update the dependency: https://github.com/p-tombez/connector-interfaces/pull/1

p-tombez committed:
Merge pull request #1 from leemannd/update-dep

IMP connector_importer_product update cryptography dep

cyrilmanuel commented:
superseeded : https://github.com/OCA/connector-interfaces/pull/182
